hackathon_site/event/api_views.py

```diff
@@ -66,3 +81,57 @@ def post(self, request, *args, **kwargs):
         response_serializer = TeamSerializer(profile.team)
         response_data = response_serializer.data
         return Response(data=response_data, status=status.HTTP_201_CREATED,)
+
+
+class JoinTeamView(generics.GenericAPIView, mixins.RetrieveModelMixin):
+    permission_classes = [UserHasProfile]
+    serializer_class = TeamSerializer
+
+    def clean(self):
+        if not is_registration_open():
+            raise forms.ValidationError(
+                _("You cannot change teams after registration has closed."),
+                code="registration_closed",
+            )
+
+        return super().clean()
+
+    @transaction.atomic
+    def post(self, request, team_code, *args, **kwargs):
+
+        profile = request.user.profile
+        current_team = profile.team
+
+        try:
+            team = Team.objects.get(team_code=team_code)
+        except:
+            raise ValidationError(
+                {"detail": "Team does not exist!"}, code=status.HTTP_400_BAD_REQUEST
+            )
+
+        if not team.profiles.exists():
+            raise ValidationError(
+                {"detail": "Cannot join a fully empty team!"},
```

Comment:
Why not? Imo this isn't a check we need since there's no harm in joining an empty team. It shouldn't happen anyway since we clean up teams when they get empty.

Reply:
removed